🧹  Resolve XML comment warnings for C# project

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Suggestion

During the compilation of the the C# project in bindings, there are errors related specifically to XML comments. These should be easily resolved with some `/// summary` tags. The descriptions can be pulled from the python repo or the underlying c code, whichever is easier. 

These can be seen in this [example run](https://github.com/microsoft/electionguard-cpp/pull/215/checks?check_run_id=3953853235) under `windows-2019-gcc-9.2.0` or `windows-2019-msvc-latest`. 

### Examples
The following warnings: 
>  warning CS1570: XML comment has badly formed XML -- 'End tag 'summary' does not match the start tag 'sumary'.'

> warning CS1572: XML comment has a param tag for 'nonceSeed', but there is no parameter by that name

> warning CS1573: Parameter 'verifyProofs' has no matching param tag in the XML comment...

> warning CS1591: Missing XML comment for publicly visible type or member 

### Possible Implementation

Add XML comments for the c sharp members

https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs1570
https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs1572
https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs1573
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs1591

### Anything else?

Replaces Issue #188

--- a/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/Collections.cs
+++ b/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/Collections.cs
@@ -77,7 +77,9 @@ namespace ElectionGuard
             return Marshal.PtrToStringAnsi(value);
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 

--- a/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/Encrypt.cs
+++ b/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/Encrypt.cs
@@ -16,16 +16,19 @@ namespace ElectionGuard
     /// The encryption device is a stateful container that represents abstract hardware
     /// authorized to participate in a specific election.
     ///
-    /// <param name="deviceUuid">a unique identifier tied to the device hardware</param>
-    /// <param name="sessionUuid">a unique identifier tied to the runtime session</param>
-    /// <param name="launchCode">a unique identifer tied to the election</param>
-    /// <param name="location">an arbitrary string meaningful to the external system
-    ///                        such as a friendly name, description, or some other value</param>
     /// </summary>
     public class EncryptionDevice : DisposableBase
     {
         internal unsafe NativeEncryptionDevice Handle;
 
+        /// <summary>
+        /// Create a new EncryptionDevice
+        /// </summary>
+        /// <param name="deviceUuid">a unique identifier tied to the device hardware</param>
+        /// <param name="sessionUuid">a unique identifier tied to the runtime session</param>
+        /// <param name="launchCode">a unique identifer tied to the election</param>
+        /// <param name="location">an arbitrary string meaningful to the external system
+        ///                        such as a friendly name, description, or some other value</param>
         public unsafe EncryptionDevice(
             ulong deviceUuid,
             ulong sessionUuid,
@@ -37,13 +40,20 @@ namespace ElectionGuard
             status.ThrowIfError();
         }
 
+        /// <summary>
+        /// Get a new hash value
+        ///
+        /// <return>An `ElementModQ`</return>
+        /// </summary>
         public unsafe ElementModQ GetHash()
         {
             var status = NativeInterface.EncryptionDevice.GetHash(Handle, out NativeElementModQ value);
             return new ElementModQ(value);
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -66,6 +76,12 @@ namespace ElectionGuard
     {
         internal unsafe NativeEncryptionMediator Handle;
 
+        /// <summary>
+        /// Create an `EncryptionMediator` object
+        /// </summary>
+        /// <param name="manifest"></param>
+        /// <param name="context"></param>
+        /// <param name="device"></param>
         public unsafe EncryptionMediator(
             InternalManifest manifest,
             CiphertextElectionContext context,
@@ -122,7 +138,9 @@ namespace ElectionGuard
 
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -132,11 +150,17 @@ namespace ElectionGuard
         }
     }
 
+    /// <summary>
+    /// Metadata for encryption
+    ///
+    /// The encrypt object is used for encrypting ballots.
+    ///
+    /// </summary>
     public class Encrypt
     {
         /// <summary>
         /// Encrypt a specific `BallotSelection` in the context of a specific `BallotContest`
-        ///
+        /// </summary>
         /// <param name="plaintext">the selection in the valid input form</param>
         /// <param name="description">the `SelectionDescription` from the `ContestDescription`
         ///                           which defines this selection's structure</param>
@@ -145,10 +169,8 @@ namespace ElectionGuard
         /// <param name="nonceSeed">an `ElementModQ` used as a header to seed the `Nonce` generated
         ///                          for this selection. this value can be (or derived from) the
         ///                          Contest nonce, but no relationship is required</param>
-        /// <param name="isPlaceholder">specifies if this is a placeholder selection</param>
         /// <param name="shouldVerifyProofs">specify if the proofs should be verified prior to returning (default True)</param>
         /// <returns>A `CiphertextBallotSelection`</returns>
-        /// </summary>
         public static unsafe CiphertextBallotSelection Selection(
             PlaintextBallotSelection plaintext,
             SelectionDescription description,
@@ -173,6 +195,7 @@ namespace ElectionGuard
         /// It will fill missing selections for a contest with `False` values, and generate `placeholder`
         /// selections to represent the number of seats available for a given contest.  By adding `placeholder`
         /// votes
+        /// </summary>
         /// <param name="plaintext">the selection in the valid input form</param>
         /// <param name="description">the `ContestDescriptionWithPlaceholders` from the `ContestDescription`
         ///                           which defines this contest's structure</param>
@@ -183,7 +206,6 @@ namespace ElectionGuard
         ///                          Ballot nonce, but no relationship is required</param>
         /// <param name="shouldVerifyProofs">specify if the proofs should be verified prior to returning (default True)</param>
         /// <returns>A `CiphertextBallotContest`</returns>
-        /// </summary>
         public static unsafe CiphertextBallotContest Contest(
             PlaintextBallotContest plaintext,
             ContestDescription description,
@@ -211,7 +233,7 @@ namespace ElectionGuard
         ///
         /// This method also allows for ballots to exclude passing contests for which the voter made no selections.
         /// It will fill missing contests with `False` selections and generate `placeholder` selections that are marked `True`.
-        ///
+        /// </summary>
         /// <param name="ballot">the selection in the valid input form</param>
         /// <param name="internalManifest">the `InternalManifest` which defines this ballot's structure</param>
         /// <param name="context">all the cryptographic context for the election</param>
@@ -220,7 +242,6 @@ namespace ElectionGuard
         ///                     if this value is not provided, the secret generating mechanism of the OS provides its own</param>
         /// <param name="shouldVerifyProofs">specify if the proofs should be verified prior to returning (default True)</param>
         /// <returns>A `CiphertextBallot`</returns>
-        /// </summary>
         public static unsafe CiphertextBallot Ballot(
             PlaintextBallot ballot,
             InternalManifest internalManifest,
@@ -262,7 +283,7 @@ namespace ElectionGuard
         ///
         /// This version of the encrypt method returns a `compact` version of the ballot that includes a minimal representation
         /// of the plaintext ballot along with the crypto parameters that are required to expand the ballot
-        ///
+        /// </summary>
         /// <param name="ballot">the selection in the valid input form</param>
         /// <param name="internalManifest">the `InternalManifest` which defines this ballot's structure</param>
         /// <param name="context">all the cryptographic context for the election</param>
@@ -271,7 +292,6 @@ namespace ElectionGuard
         ///                     if this value is not provided, the secret generating mechanism of the OS provides its own</param>
         /// <param name="shouldVerifyProofs">specify if the proofs should be verified prior to returning (default True)</param>
         /// <returns>A `CiphertextBallot`</returns>
-        /// </summary>
         public static unsafe CompactCiphertextBallot CompactBallot(
             PlaintextBallot ballot,
             InternalManifest internalManifest,

--- a/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/Group.cs
+++ b/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/Group.cs
@@ -11,6 +11,9 @@ namespace ElectionGuard
     /// </summary>
     public class ElementModP : DisposableBase
     {
+        /// <summary>
+        /// Number of 64-bit ints that make up the 4096-bit prime
+        /// </summary>
         public static readonly ulong MAX_SIZE = 64;
 
         /// <Summary>
@@ -19,6 +22,10 @@ namespace ElectionGuard
         public ulong[] Data { get { return GetNative(); } internal set { NewNative(value); } }
         internal unsafe NaiveElementModP Handle;
 
+        /// <summary>
+        /// Creates a `ElementModP` object
+        /// </summary>
+        /// <param name="data">the data used to initialize the `ElementModP`</param>
         public ElementModP(ulong[] data)
         {
             try
@@ -36,7 +43,9 @@ namespace ElectionGuard
             Handle = handle;
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -114,6 +123,9 @@ namespace ElectionGuard
     /// </summary>
     public class ElementModQ : DisposableBase
     {
+        /// <summary>
+        /// Number of 64-bit ints that make up the 256-bit prime
+        /// </summary>
         public static readonly ulong MAX_SIZE = 4;
 
         /// <Summary>
@@ -122,6 +134,10 @@ namespace ElectionGuard
         public ulong[] Data { get { return GetNative(); } internal set { NewNative(value); } }
         internal unsafe NaiveElementModQ Handle;
 
+        /// <summary>
+        /// Create a `ElementModQ`
+        /// </summary>
+        /// <param name="data">data used to initialize the `ElementModQ`</param>
         public ElementModQ(ulong[] data)
         {
             try
@@ -134,6 +150,11 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Create a `ElementModQ`
+        /// </summary>
+        /// <param name="hex">string representing the hex bytes of the initializationd data</param>
+        /// <param name="uncheckedInput">if data is checked or not</param>
         public ElementModQ(string hex, bool uncheckedInput = false)
         {
             var status = uncheckedInput ?
@@ -161,7 +182,9 @@ namespace ElectionGuard
             return value;
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -220,8 +243,14 @@ namespace ElectionGuard
         }
     }
 
+    /// <summary>
+    /// Static class of constant values used in the encryption system
+    /// </summary>
     public static class Constants
     {
+        /// <summary>
+        /// G in Hacl_Bignum4096 format
+        /// </summary>
         public unsafe static ElementModP G
         {
             get
@@ -235,6 +264,9 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Max P value in Hacl_Bignum4096 format
+        /// </summary>
         public unsafe static ElementModP P
         {
             get
@@ -248,6 +280,9 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// zero as data for `ElementModP`
+        /// </summary>
         public unsafe static ElementModP ZERO_MOD_P
         {
             get
@@ -261,6 +296,9 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// one as data for `ElementModP`
+        /// </summary>
         public unsafe static ElementModP ONE_MOD_P
         {
             get
@@ -274,6 +312,9 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// two as data for `ElementModP`
+        /// </summary>
         public unsafe static ElementModP TWO_MOD_P
         {
             get
@@ -287,6 +328,9 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Max Q value in Hacl_Bignum256 format
+        /// </summary>
         public unsafe static ElementModQ Q
         {
             get
@@ -300,6 +344,9 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// zero as data for `ElementModQ`
+        /// </summary>
         public unsafe static ElementModQ ZERO_MOD_Q
         {
             get
@@ -313,6 +360,9 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// one as data for `ElementModQ`
+        /// </summary>
         public unsafe static ElementModQ ONE_MOD_Q
         {
             get
@@ -326,6 +376,9 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// two as data for `ElementModQ`
+        /// </summary>
         public unsafe static ElementModQ TWO_MOD_Q
         {
             get

--- a/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/Hash.cs
+++ b/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/Hash.cs
@@ -1,8 +1,14 @@
 ﻿using System;
 namespace ElectionGuard.Encryption
 {
+    /// <summary>
+    /// Class for wrapping hashing methods
+    /// </summary>
     public class Hash
     {
+        /// <summary>
+        /// Create a Hash object
+        /// </summary>
         public Hash()
         {
         }

--- a/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/Manifest.cs
+++ b/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/Manifest.cs
@@ -71,6 +71,11 @@ namespace ElectionGuard
             Handle = handle;
         }
 
+        /// <summary>
+        /// Creating a new `AnnotatedString`
+        /// </summary>
+        /// <param name="annotation">annotation for new string</param>
+        /// <param name="value">string value</param>
         public unsafe AnnotatedString(string annotation, string value)
         {
             var status = NativeInterface.AnnotatedString.New(annotation, value, out Handle);
@@ -80,7 +85,9 @@ namespace ElectionGuard
             }
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -160,6 +167,11 @@ namespace ElectionGuard
             Handle = handle;
         }
 
+        /// <summary>
+        /// Create a new `Language` object
+        /// </summary>
+        /// <param name="value">value to represent language</param>
+        /// <param name="language">string with language info</param>
         public unsafe Language(string value, string language)
         {
             var status = NativeInterface.Language.New(value, language, out Handle);
@@ -169,7 +181,9 @@ namespace ElectionGuard
             }
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -225,6 +239,10 @@ namespace ElectionGuard
             Handle = handle;
         }
 
+        /// <summary>
+        /// Create an `InternationalizedText` object
+        /// </summary>
+        /// <param name="text">array of text for languages</param>
         public unsafe InternationalizedText(Language[] text)
         {
             IntPtr[] nativeText = new IntPtr[text.Length];
@@ -241,7 +259,9 @@ namespace ElectionGuard
             }
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -250,6 +270,11 @@ namespace ElectionGuard
             Handle = null;
         }
 
+        /// <summary>
+        /// Gets text from array of languages
+        /// </summary>
+        /// <param name="index">index to use to get `Language`</param>
+        /// <returns>`Language` object</returns>
         public unsafe Language GetTextAt(ulong index)
         {
             var status = NativeInterface.InternationalizedText.GetTextAtIndex(
@@ -339,6 +364,10 @@ namespace ElectionGuard
             Handle = handle;
         }
 
+        /// <summary>
+        /// Create a `ContactInformation` object
+        /// </summary>
+        /// <param name="name">name of the contact</param>
         public unsafe ContactInformation(string name)
         {
             var status = NativeInterface.ContactInformation.New(name, out Handle);
@@ -348,7 +377,9 @@ namespace ElectionGuard
             }
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -498,6 +529,12 @@ namespace ElectionGuard
             Handle = handle;
         }
 
+        /// <summary>
+        /// Create a `GeopoliticalUnit` object
+        /// </summary>
+        /// <param name="objectId">string to identify the unit</param>
+        /// <param name="name">name of the unit</param>
+        /// <param name="reportingUnitType">type of geopolitical unit</param>
         public unsafe GeopoliticalUnit(
             string objectId, string name, ReportingUnitType reportingUnitType)
         {
@@ -509,7 +546,9 @@ namespace ElectionGuard
             }
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -601,6 +640,11 @@ namespace ElectionGuard
             Handle = handle;
         }
 
+        /// <summary>
+        /// Create a `BallotStyle` object
+        /// </summary>
+        /// <param name="objectId">string to identify the `BallotStyle`</param>
+        /// <param name="gpUnitIds">array of objectIds for the `GeopoliticalUnit` for this `BallotStyle`</param>
         public unsafe BallotStyle(string objectId, string[] gpUnitIds)
         {
             var status = NativeInterface.BallotStyle.New(objectId, gpUnitIds, (ulong)gpUnitIds.Length, out Handle);
@@ -610,7 +654,9 @@ namespace ElectionGuard
             }
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -776,6 +822,10 @@ namespace ElectionGuard
             Handle = handle;
         }
 
+        /// <summary>
+        /// Create a `Party` object
+        /// </summary>
+        /// <param name="objectId">string to identify the object</param>
         public unsafe Party(string objectId)
         {
             var status = NativeInterface.Party.New(objectId, out Handle);
@@ -785,6 +835,14 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Create a `Party` object
+        /// </summary>
+        /// <param name="objectId">string to identify the object</param>
+        /// <param name="name">name of the party</param>
+        /// <param name="abbreviation">abbreviation for the object</param>
+        /// <param name="color">string for the name of the color used</param>
+        /// <param name="logoUri">string for the uri for the logo</param>
         public unsafe Party(
             string objectId, InternationalizedText name,
             string abbreviation, string color, string logoUri)
@@ -797,7 +855,9 @@ namespace ElectionGuard
             }
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -936,6 +996,11 @@ namespace ElectionGuard
             Handle = handle;
         }
 
+        /// <summary>
+        /// Create a `Candidate` object
+        /// </summary>
+        /// <param name="objectId">string for the identity of the object</param>
+        /// <param name="isWriteIn">is the candidate a write in</param>
         public unsafe Candidate(string objectId, bool isWriteIn)
         {
             var status = NativeInterface.Candidate.New(objectId, isWriteIn, out Handle);
@@ -945,6 +1010,12 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Create a `Candidate` object
+        /// </summary>
+        /// <param name="objectId">string for the identity of the object</param>
+        /// <param name="partyId">string identifying the party for the candidate</param>
+        /// <param name="isWriteIn">is the candidate a write in</param>
         public unsafe Candidate(string objectId, string partyId, bool isWriteIn)
         {
             var status = NativeInterface.Candidate.New(
@@ -955,6 +1026,14 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Create a `Candidate` object
+        /// </summary>
+        /// <param name="objectId">string for the identity of the object</param>
+        /// <param name="name">name of the candidate</param>
+        /// <param name="partyId">string identifying the party for the candidate</param>
+        /// <param name="imageUri">string for uir for image of candidate</param>
+        /// <param name="isWriteIn">is the candidate a write in</param>
         public unsafe Candidate(
             string objectId, InternationalizedText name,
             string partyId, string imageUri, bool isWriteIn)
@@ -967,7 +1046,9 @@ namespace ElectionGuard
             }
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -1070,6 +1151,12 @@ namespace ElectionGuard
             Handle = handle;
         }
 
+        /// <summary>
+        /// Create a `SelectionDescription` object
+        /// </summary>
+        /// <param name="objectId">string identifying the object</param>
+        /// <param name="candidateId">string identifying the candidate</param>
+        /// <param name="sequenceOrder">the number of the item for sequence order</param>
         public unsafe SelectionDescription(
             string objectId, string candidateId, ulong sequenceOrder)
         {
@@ -1081,7 +1168,9 @@ namespace ElectionGuard
             }
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -1284,6 +1373,16 @@ namespace ElectionGuard
             Handle = handle;
         }
 
+        /// <summary>
+        /// Create a `ContestDescription` object
+        /// </summary>
+        /// <param name="objectId">string identifying object</param>
+        /// <param name="electoralDistrictId">string identifying electoral district</param>
+        /// <param name="sequenceOrder">the sequence order to show this in</param>
+        /// <param name="voteVariation">vote variation type</param>
+        /// <param name="numberElected">the number of elected</param>
+        /// <param name="name">string for name of the contest</param>
+        /// <param name="selections">array of `SelectionDescription`</param>
         public unsafe ContestDescription(
             string objectId, string electoralDistrictId, ulong sequenceOrder,
             VoteVariationType voteVariation, ulong numberElected, string name,
@@ -1306,6 +1405,19 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Create a `ContestDescription` object
+        /// </summary>
+        /// <param name="objectId">string identifying object</param>
+        /// <param name="electoralDistrictId">string identifying electoral district</param>
+        /// <param name="sequenceOrder">the sequence order to show this in</param>
+        /// <param name="voteVariation">vote variation type</param>
+        /// <param name="numberElected">number of elected</param>
+        /// <param name="votesAllowed">number of votes allowed</param>
+        /// <param name="name">string for name of the contest</param>
+        /// <param name="ballotTitle">international string for the ballot title</param>
+        /// <param name="ballotSubtitle">international string for the ballot title</param>
+        /// <param name="selections">array of `SelectionDescription`</param>
         public unsafe ContestDescription(
             string objectId, string electoralDistrictId, ulong sequenceOrder,
             VoteVariationType voteVariation, ulong numberElected, ulong votesAllowed,
@@ -1330,6 +1442,17 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Create a `ContestDescription` object
+        /// </summary>
+        /// <param name="objectId">string identifying object</param>
+        /// <param name="electoralDistrictId">string identifying electoral district</param>
+        /// <param name="sequenceOrder">the sequence order to show this in</param>
+        /// <param name="voteVariation">vote variation type</param>
+        /// <param name="numberElected">number of elected</param>
+        /// <param name="name">string for name of the contest</param>
+        /// <param name="selections">array of `SelectionDescription`</param>
+        /// <param name="primaryPartyIds">array of strings for `PartyIds`</param>
         public unsafe ContestDescription(
             string objectId, string electoralDistrictId, ulong sequenceOrder,
             VoteVariationType voteVariation, ulong numberElected, string name,
@@ -1353,6 +1476,20 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Create a `ContestDescription` object
+        /// </summary>
+        /// <param name="objectId">string identifying object</param>
+        /// <param name="electoralDistrictId">string identifying electoral district</param>
+        /// <param name="sequenceOrder">the sequence order to show this in</param>
+        /// <param name="voteVariation">vote variation type</param>
+        /// <param name="numberElected">number of elected</param>
+        /// <param name="votesAllowed">number of votes allowed</param>
+        /// <param name="name">string for name of the contest</param>
+        /// <param name="ballotTitle">international string for the ballot title</param>
+        /// <param name="ballotSubtitle">international string for the ballot title</param>
+        /// <param name="selections">array of `SelectionDescription`</param>
+        /// <param name="primaryPartyIds">array of strings for `PartyIds`</param>
         public unsafe ContestDescription(
             string objectId, string electoralDistrictId, ulong sequenceOrder,
             VoteVariationType voteVariation, ulong numberElected, ulong votesAllowed,
@@ -1378,7 +1515,9 @@ namespace ElectionGuard
             }
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -1604,6 +1743,17 @@ namespace ElectionGuard
             Handle = handle;
         }
 
+        /// <summary>
+        /// Create a `ContestDescriptionWithPlaceholders` object
+        /// </summary>
+        /// <param name="objectId">string identifying object</param>
+        /// <param name="electoralDistrictId">string identifying electoral district</param>
+        /// <param name="sequenceOrder">the sequence order to show this in</param>
+        /// <param name="voteVariation">vote variation type</param>
+        /// <param name="numberElected">the number of elected</param>
+        /// <param name="name">string for name of the contest</param>
+        /// <param name="selections">array of `SelectionDescription`</param>
+        /// <param name="placeholders">array of `SelectionDescription` to use as placeholders</param>
         public unsafe ContestDescriptionWithPlaceholders(
             string objectId, string electoralDistrictId, ulong sequenceOrder,
             VoteVariationType voteVariation, ulong numberElected, string name,
@@ -1635,6 +1785,20 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Create a `ContestDescriptionWithPlaceholders` object
+        /// </summary>
+        /// <param name="objectId">string identifying object</param>
+        /// <param name="electoralDistrictId">string identifying electoral district</param>
+        /// <param name="sequenceOrder">the sequence order to show this in</param>
+        /// <param name="voteVariation">vote variation type</param>
+        /// <param name="numberElected">the number of elected</param>
+        /// <param name="votesAllowed">number of votes allowed</param>
+        /// <param name="name">string for name of the contest</param>
+        /// <param name="ballotTitle">international string for the ballot title</param>
+        /// <param name="ballotSubtitle">international string for the ballot title</param>
+        /// <param name="selections">array of `SelectionDescription`</param>
+        /// <param name="placeholders">array of `SelectionDescription` to use as placeholders</param>
         public unsafe ContestDescriptionWithPlaceholders(
             string objectId, string electoralDistrictId, ulong sequenceOrder,
             VoteVariationType voteVariation, ulong numberElected, ulong votesAllowed,
@@ -1668,6 +1832,18 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Create a `ContestDescriptionWithPlaceholders` object
+        /// </summary>
+        /// <param name="objectId">string identifying object</param>
+        /// <param name="electoralDistrictId">string identifying electoral district</param>
+        /// <param name="sequenceOrder">the sequence order to show this in</param>
+        /// <param name="voteVariation">vote variation type</param>
+        /// <param name="numberElected">the number of elected</param>
+        /// <param name="name">string for name of the contest</param>
+        /// <param name="selections">array of `SelectionDescription`</param>
+        /// <param name="primaryPartyIds">array of strings for `PartyIds`</param>
+        /// <param name="placeholders">array of `SelectionDescription` to use as placeholders</param>
         public unsafe ContestDescriptionWithPlaceholders(
             string objectId, string electoralDistrictId, ulong sequenceOrder,
             VoteVariationType voteVariation, ulong numberElected, string name,
@@ -1701,6 +1877,21 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Create a `ContestDescriptionWithPlaceholders` object
+        /// </summary>
+        /// <param name="objectId">string identifying object</param>
+        /// <param name="electoralDistrictId">string identifying electoral district</param>
+        /// <param name="sequenceOrder">the sequence order to show this in</param>
+        /// <param name="voteVariation">vote variation type</param>
+        /// <param name="numberElected">the number of elected</param>
+        /// <param name="votesAllowed">number of votes allowed</param>
+        /// <param name="name">string for name of the contest</param>
+        /// <param name="ballotTitle">international string for the ballot title</param>
+        /// <param name="ballotSubtitle">international string for the ballot title</param>
+        /// <param name="selections">array of `SelectionDescription`</param>
+        /// <param name="primaryPartyIds">array of strings for `PartyIds`</param>
+        /// <param name="placeholders">array of `SelectionDescription` to use as placeholders</param>
         public unsafe ContestDescriptionWithPlaceholders(
             string objectId, string electoralDistrictId, ulong sequenceOrder,
             VoteVariationType voteVariation, ulong numberElected, ulong votesAllowed,
@@ -1736,7 +1927,9 @@ namespace ElectionGuard
             }
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -1966,6 +2159,10 @@ namespace ElectionGuard
             Handle = handle;
         }
 
+        /// <summary>
+        /// Creates a `Manifest` object
+        /// </summary>
+        /// <param name="json">string of json data describing the manifest</param>
         public unsafe Manifest(string json)
         {
             var status = NativeInterface.Manifest.FromJson(json, out Handle);
@@ -1975,6 +2172,11 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Creates a `Manifest` object
+        /// </summary>
+        /// <param name="data">byte array of data describing the manifest</param>
+        /// <param name="encoding">binary encoding for the data</param>
         public unsafe Manifest(byte[] data, BinarySerializationEncoding encoding)
         {
             fixed (byte* pointer = data)
@@ -1989,6 +2191,18 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Creates a `Manifest` object
+        /// </summary>
+        /// <param name="electionScopeId"></param>
+        /// <param name="electionType">election type</param>
+        /// <param name="startDate">start date for election</param>
+        /// <param name="endDate">end data for the election</param>
+        /// <param name="gpUnits">array of the `GeopoliticalUnit` for election</param>
+        /// <param name="parties">array of the `Party` for election</param>
+        /// <param name="candidates">array of the `Candidate` for election</param>
+        /// <param name="contests">array of the `ContestDescription` for election</param>
+        /// <param name="ballotStyles">array of the `BallotStyle` for election</param>
         public unsafe Manifest(
             string electionScopeId, ElectionType electionType,
             DateTime startDate, DateTime endDate,
@@ -2047,7 +2261,9 @@ namespace ElectionGuard
             }
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 
@@ -2287,6 +2503,10 @@ namespace ElectionGuard
 
         internal unsafe NativeInternalManifest Handle;
 
+        /// <summary>
+        /// Creates an `InternalManifest` object
+        /// </summary>
+        /// <param name="manifest">public manifest to copy</param>
         public unsafe InternalManifest(Manifest manifest)
         {
             var status = NativeInterface.InternalManifest.New(manifest.Handle, out Handle);
@@ -2296,6 +2516,10 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Creates an `InternalManifest` object
+        /// </summary>
+        /// <param name="json">string of json data describing the manifest</param>
         public unsafe InternalManifest(string json)
         {
             var status = NativeInterface.InternalManifest.FromJson(json, out Handle);
@@ -2305,6 +2529,11 @@ namespace ElectionGuard
             }
         }
 
+        /// <summary>
+        /// Creates an `InternalManifest` object
+        /// </summary>
+        /// <param name="data">byte array of data describing the manifest</param>
+        /// <param name="encoding">binary encoding for the data</param>
         public unsafe InternalManifest(byte[] data, BinarySerializationEncoding encoding)
         {
             fixed (byte* pointer = data)
@@ -2319,7 +2548,9 @@ namespace ElectionGuard
             }
         }
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
         protected override unsafe void DisposeUnmanaged()
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
         {
             base.DisposeUnmanaged();
 

--- a/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/NativeInterface.cs
+++ b/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/NativeInterface.cs
@@ -76,13 +76,37 @@ namespace ElectionGuard
     /// </Summary>
     public enum ElectionType
     {
+        /// <summary>
+        /// an unknown value
+        /// </summary>
         unknown = 0,
+        /// <summary>
+        /// An election that's typically held on the national day for elections.
+        /// </summary>
         general = 1,
+        /// <summary>
+        /// A primary election that's for a specific party, where voter eligibility for this contest is based on registration.
+        /// </summary>
         partisanPrimaryClosed = 2,
+        /// <summary>
+        /// A primary election that's for a specific party, where voters declare their desired party or choose in private
+        /// </summary>
         partisanPrimaryOpen = 3,
+        /// <summary>
+        /// A primary election without a specified type, such as a nonpartisan primary.
+        /// </summary>
         primary = 4,
+        /// <summary>
+        /// An election to decide a prior contest that ended with no candidate receiving a majority of the votes
+        /// </summary>
         runoff = 5,
+        /// <summary>
+        /// An election held out of sequence for special circumstances, such as to fill a vacated office.
+        /// </summary>
         special = 6,
+        /// <summary>
+        /// The election is a type that isn't listed in this enumeration. If used, include the item's custom type in an OtherType element.
+        /// </summary>
         other = 7
     };
 
@@ -92,34 +116,127 @@ namespace ElectionGuard
     /// </Summary>
     public enum ReportingUnitType
     {
+        /// <summary>
+        /// an unknown value
+        /// </summary>
         unknown = 0,
+        /// <summary>
+        /// Used to report batches of ballots that might cross precinct boundaries.
+        /// </summary>
         ballotBatch = 1,
+        /// <summary>
+        /// Used for a ballot-style area that's generally composed of precincts.
+        /// </summary>
         ballotStyleArea = 2,
+        /// <summary>
+        /// Used as a synonym for a county.
+        /// </summary>
         borough = 3,
+        /// <summary>
+        /// Used for a city that reports results or for the district that encompasses it.
+        /// </summary>
         city = 4,
+        /// <summary>
+        /// Used for city council districts.
+        /// </summary>
         cityCouncil = 5,
+        /// <summary>
+        /// Used for one or more precincts that have been combined for the purposes of reporting. 
+        /// If the term ward is used interchangeably with combined precinct, use combined-precinct 
+        /// for the ReportingUnitType.
+        /// </summary>
         combinedPrecinct = 6,
+        /// <summary>
+        /// Used for national legislative body districts.
+        /// </summary>
         congressional = 7,
+        /// <summary>
+        /// Used for a country.
+        /// </summary>
         country = 8,
+        /// <summary>
+        /// Used for a county or for the district that encompasses it. 
+        /// Synonymous with borough and parish in some localities.
+        /// </summary>
         county = 9,
+        /// <summary>
+        /// Used for county council districts.
+        /// </summary>
         countyCouncil = 10,
+        /// <summary>
+        /// Used for a dropbox for absentee ballots.
+        /// </summary>
         dropBox = 11,
+        /// <summary>
+        /// Used for judicial districts.
+        /// </summary>
         judicial = 12,
+        /// <summary>
+        /// Used as applicable for various units such as towns, 
+        /// townships, villages that report votes, or for the district 
+        /// that encompasses them.
+        /// </summary>
         municipality = 13,
+        /// <summary>
+        /// Used for a polling place.
+        /// </summary>
         polling_place = 14,
+        /// <summary>
+        /// Used if the terms for ward or district are used interchangeably with precinct.
+        /// </summary>
         precinct = 15,
+        /// <summary>
+        /// Used for a school district.
+        /// </summary>
         school = 16,
+        /// <summary>
+        /// Used for a special district.
+        /// </summary>
         special = 17,
+        /// <summary>
+        /// Used for splits of precincts.
+        /// </summary>
         splitPrecinct = 18,
+        /// <summary>
+        /// Used for a state or for the district that encompasses it.
+        /// </summary>
         state = 19,
+        /// <summary>
+        /// Used for a state house or assembly district.
+        /// </summary>
         stateHouse = 20,
+        /// <summary>
+        /// Used for a state senate district.
+        /// </summary>
         stateSenate = 21,
+        /// <summary>
+        /// Used for type of municipality that reports votes or for the district that encompasses it.
+        /// </summary>
         township = 22,
+        /// <summary>
+        /// Used for a utility district.
+        /// </summary>
         utility = 23,
+        /// <summary>
+        /// Used for a type of municipality that reports votes or for the district that encompasses it.
+        /// </summary>
         village = 24,
+        /// <summary>
+        /// Used for a vote center.
+        /// </summary>
         voteCenter = 25,
+        /// <summary>
+        /// Used for combinations or groupings of precincts or other units.
+        /// </summary>
         ward = 26,
+        /// <summary>
+        /// Used for a water district.
+        /// </summary>
         water = 27,
+        /// <summary>
+        /// Used for other types of reporting units that aren't included in this enumeration. 
+        /// If used, provide the item's custom type in an OtherType element.
+        /// </summary>
         other = 28,
     };
 
@@ -129,18 +246,65 @@ namespace ElectionGuard
     /// </Summary>
     public enum VoteVariationType
     {
+        /// <summary>
+        /// an unknown value
+        /// </summary>
         unknown = 0,
+        /// <summary>
+        /// Each voter can select up to one option.
+        /// </summary>
         one_of_m = 1,
+        /// <summary>
+        /// Approval voting, where each voter can select as many options as desired.
+        /// </summary>
         approval = 2,
+        /// <summary>
+        /// Borda count, where each voter can rank the options, and the rankings are assigned point values.
+        /// </summary>
         borda = 3,
+        /// <summary>
+        /// Cumulative voting, where each voter can distribute their vote to up to N options.
+        /// </summary>
         cumulative = 4,
+        /// <summary>
+        /// A 1-of-m method where the winner needs more than 50% of the vote to be elected.
+        /// </summary>
         majority = 5,
+        /// <summary>
+        /// A method where each voter can select up to N options.
+        /// </summary>
         n_of_m = 6,
+        /// <summary>
+        /// A 1-of-m method where the option with the most votes is elected, regardless of 
+        /// whether the option has more than 50% of the vote.
+        /// </summary>
         plurality = 7,
+        /// <summary>
+        /// A proportional representation method, which is any system that elects winners 
+        /// in proportion to the total vote. For the single transferable vote (STV) method, 
+        /// use rcv instead.
+        /// </summary>
         proportional = 8,
+        /// <summary>
+        /// Range voting, where each voter can select a score for each option.
+        /// </summary>
         range = 9,
+        /// <summary>
+        /// Ranked choice voting (RCV), where each voter can rank the options, and the 
+        /// ballots are counted in rounds. Also known as instant-runoff voting (IRV) and 
+        /// the single transferable vote (STV).
+        /// </summary>
         rcv = 10,
+        /// <summary>
+        /// A 1-of-m method where the winner needs more than some predetermined fraction 
+        /// of the vote to be elected, and where the fraction is more than 50%. For example, 
+        /// the winner might need three-fifths or two-thirds of the vote.
+        /// </summary>
         super_majority = 11,
+        /// <summary>
+        /// The vote variation is a type that isn't included in this enumeration. 
+        /// If used, provide the item's custom type in an OtherType element.
+        /// </summary>
         other = 12
     };
 

--- a/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/Utilities.cs
+++ b/bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/Utilities.cs
@@ -1,8 +1,14 @@
 ﻿using System;
 namespace ElectionGuard.Encryption
 {
+    /// <summary>
+    /// Class for wrapping around utility functions
+    /// </summary>
     public class Utilities
     {
+        /// <summary>
+        /// Create Utilities object
+        /// </summary>
         public Utilities()
         {
         }

